client/blocks/upload-image/index.jsx

```diff
@@ -0,0 +1,205 @@
+import Spinner from 'components/spinner';
+import ImageEditor from 'blocks/image-editor';
+import DropZone from 'components/drop-zone';
+
+class UploadImage extends Component {
+	state = {
+		isEditingImage: false,
+		uploadedImage: null,
+		editedImage: null,
+	};
+
+	static propTypes = {
+		isUploading: PropTypes.bool,
+		// Additional props passed to ImageEditor component. See blocks/image-editor.
+		imageEditorProps: PropTypes.object,
+		texts: PropTypes.object,
+		onImageEditorDone: PropTypes.func,
+		additionalImageEditorClasses: PropTypes.string,
+		additionalClasses: PropTypes.string,
+		placeholderContent: PropTypes.element,
+		uploadingContent: PropTypes.element,
+	};
+
+	static defaultProps = {
+		imageEditorProps: {
+			allowedAspectRatios: [ AspectRatios.ASPECT_1X1 ],
+		},
+		texts: {},
+		backgroundContent: null,
+		onImageEditorDone: noop,
+		isUploading: false,
+	};
+
+	receiveFiles = ( files ) => {
+		const extension = path.extname( files[ 0 ].name )
+			.toLowerCase()
+			.substring( 1 );
+
+		if ( ALLOWED_FILE_EXTENSIONS.indexOf( extension ) === -1 ) {
+			return;
+		}
+
+		const imageObjectUrl = URL.createObjectURL( files[ 0 ] );
+
+		this.setState( {
+			isEditingImage: true,
+			uploadedImage: imageObjectUrl,
+		} );
+	};
+
+	onImageEditorDone = ( error, imageBlob ) => {
+		this.hideImageEditor();
+
+		if ( error ) {
+			return;
+		}
+
+		this.setState( { editedImage: URL.createObjectURL( imageBlob ) } );
+
+		this.props.onImageEditorDone( imageBlob );
+	};
+
+	hideImageEditor = () => {
+		const {
+			resetAllImageEditorState: resetAllImageEditorStateAction
+		} = this.props;
+
+		resetAllImageEditorStateAction();
+
+		URL.revokeObjectURL( this.state.uploadedImage );
+
+		this.setState( {
+			isEditingImage: false,
+			uploadedImage: false,
+		} );
+	};
+
+	renderImageEditor() {
+		const {
+			isEditingImage,
+			uploadedImage,
+		} = this.state;
+
+		if ( ! isEditingImage ) {
+			return null;
+		}
+
+		const {
+			additionalImageEditorClasses,
+			imageEditorProps,
+			texts,
+		} = this.props;
+
+		const classes = classnames( 'upload-image-modal', additionalImageEditorClasses );
+
+		return (
+			<Dialog
+				additionalClassNames={ classes }
+				isVisible={ true }
+			>
+				<ImageEditor
+					{ ...imageEditorProps }
+					media={ { src: uploadedImage } }
+					onDone={ this.onImageEditorDone }
+					onCancel={ this.hideImageEditor }
+					doneButtonText={ texts.doneButtonText ? texts.doneButtonText : 'Done' }
+				/>
+			</Dialog>
+		);
+	}
+
+	render() {
+		const {
+			children,
+			isUploading,
+			translate,
+			texts,
+			additionalClasses,
+		} = this.props;
+
+		let {
+			placeholderContent,
+			uploadingContent,
+		} = this.props;
+
+		if ( typeof placeholderContent === 'undefined' ) {
+			placeholderContent = (
+				<div className="upload-image__placeholder">
+						<Gridicon icon="add-image" size={ 36 } />
+					<span>
+						{ texts.addAnImage ? texts.addAnImage : translate( 'Add an image' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **75** times:
`translate( 'Add an Image' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).